src/std/misc/string.ss

```diff
@@ -215,3 +219,24 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; When str receives a single argument, the value is turned into a string.
+;; When it receives several arguments, which all must be strings, the
+;; strings are concatenated into one. When called without an argument an
+;; empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def str
+  (case-lambda
+    (() "")
+    ((v) (let (s (format (if (representable? v) "~r" "~a") v))
+           (if (and (number? v) (string-suffix? "." s))
+             (string-append s "0")
+             s)))
+    ((s . xs) (string-concatenate (cons s xs)))))
```

Comment:
Why not apply `str` to all arguments and then `string-concatenate`? Seems like an arbitrary restriction, and it is also just a synonym for `string-append`.

Reply:
for efficiency reasons the string conversion procedure of on argument should be reified out of the case-lambda (otherwise the map dispatch will go through the case-lambda dispatch).

Reply:
Is there anything to gain here using case-lambda or should I just combine it into one procedure?

Reply:
Probably not; you can write it as:
```
(def (str . args) (append-strings (map str1 args)))
```

where `str1` is the single element conversion.

Reply:
Maybe there is some gain for optimizing the single argument case as direct dispatch to `str1`; this will avoid creating the intermediate list when direct dispatched.
So:
```
(def* str
  ((x) (str1 x))
  (xs (append-strings (map str1 xs))))
```

Reply:
Would `call-with-output-string` with `write-string` not even be better because we skip the intermediate list?

Reply:
No, that creates an intermediate port and uses slower procedures!

Reply:
Oh, we are already making a port because of format, nm.

Reply:
It might make sense indeed to open a string port and use `fprintf` instead of `format`; as it is we are making a string port for each argument.
We can still specially handle the single argument case and call format.